DESIGN.md

```diff
@@ -0,0 +1,50 @@
+### Define Your Schema
+
+To define the schema, you can use MarkdownDB's flexible schema definition syntax. Below is an example for a blog post with a mandatory `date` field:
+
+```javascript
+// markdowndb.config.js
+
+module.exports = {
+  schemas: {
+    post: {
+      date: {
+        type: "string",
+        required: true,
+        validate: (fileObject) => {
+          const dateFormat = /\d{4}-\d{2}-\d{2}/; // YYYY-MM-DD
+          if (!dateFormat.test(fileObject.date)) {
+            return {
+              status: false,
+              message: "Invalid date format. Please use YYYY-MM-DD.",
+            };
+          }
+          return {
+            status: true,
+          };
+        },
+        compute(fileObject, ast) {
```

Comment:
i think we want to clarify how this part works:

1. Defining "plugins" that transform stuff
2. How to configure which ones are used ...

IMO i'd just do it the remark approach where you have some kind of configuration moment where you keep adding the items you want like:

```
const file = await unified()
  .use(remarkParse)
  .use(remarkRehype)
  .use(rehypeSanitize)
  .use(rehypeStringify)
  .process('# Hello, Neptune!')
```